File: Frontend/types/pengaduan.ts
export type KategoriPengaduan = 'infrastruktur' | 'layanan' | 'lingkungan' | 'lainnya';
export type StatusPengaduan = 'diterima' | 'ditindaklanjuti' | 'selesai';

export interface Pengaduan {
  id: string;
  nomorTiket: string;
  kategori: KategoriPengaduan;
  judul: string;
  deskripsi: string;
  lokasi?: string;
  lampiran?: string[];
  status: StatusPengaduan;
  tanggapanAdmin?: string;
  pelaporId: string;
  pelaporNama: string;
  dibuatPada: string;
  diperbaruiPada: string;
}

export interface SubmitPengaduanInput {
  kategori: KategoriPengaduan;
  judul: string;
  deskripsi: string;
  lokasi?: string;
}

File: Backend/internal/domain/pengaduan.go
package domain

import "time"

type StatusPengaduan string

const (
	PengaduanDiterima        StatusPengaduan = "diterima"
	PengaduanDitindaklanjuti StatusPengaduan = "ditindaklanjuti"
	PengaduanSelesai         StatusPengaduan = "selesai"
)

type Pengaduan struct {
	ID, NomorTiket, PelaporID, Kategori, Judul, Deskripsi string
	Lokasi                                                *string
	Status                                                StatusPengaduan
	TanggapanAdmin                                        *string
	ChangedBy                                             *string
	PelaporNama                                           string
	CreatedAt, UpdatedAt                                  time.Time
}
